### DISCUSSION TITLE
Capstone-Project-Telebot

### DISCUSSION THREAD

**Discussion by adrianpratamasoe** [teamalgoritma/community#47] [Q&A] [answered]
Nama: Adrian 
Pertanyaan: Error di Checkpoint 1, A request to the Telegram API was unsuccessful. Error Code 400. Description: Bad Request: chat not found
![Picture1](https://user-images.githubusercontent.com/40779899/163101735-b937f450-423e-4e8d-a160-9b1b28f6909b.png)

**Comment by kevbow** [comment, marked as answer]
Halo Pak Adrian,

Sepertinya kendala yang Anda hadapi karena script bot_local.py Anda belum berjalan. Hal ini karena Anda masih menggunakan powershell pada terminal Anda. Silakan Anda ubah ke mode Command Prompt menggunakan step di bawah ini:
1. Pada sebelah kanan terminal, terdapat icon (+) dan juga icon panah ke bawah
2. Klik tanda panah ke bawah tsb, kemudian pilih Command Prompt
Ilustrasi gambar :
![image](https://user-images.githubusercontent.com/68848168/163301240-be82b9fc-a137-4907-8108-d184f5fe1c4d.png)

**Comment by adrianpratamasoe** [comment]
![image](https://user-images.githubusercontent.com/40779899/163330822-cce09527-e978-4cfc-b32e-ae19dddfb5b1.png)
Ini sepertinya botnya masih tidak terconnect, walaupun sudah menggunakan kernel yang sesuai dan menggunakan command prompt.
Apakah ini masalah koneksi ataukah bisa masalah lain?
![image](https://user-images.githubusercontent.com/40779899/163356827-764337f6-a3d2-4c47-8ea4-184dbd1537ca.png)

**Comment by kevbow** [comment]
Halo Pak Adrian, 

Betul, sepertinya bot belum terkoneksi. Ada baiknya Anda coba membuat ulang bot nya menggunakan BotFather dan coba mengganti token yang Anda gunakan dengan token bot baru tersebut.

**Comment by adrianpratamasoe** [comment]
![image](https://user-images.githubusercontent.com/40779899/163533269-be2edd1d-2353-4390-a31f-8e65ab48d030.png)
Sepertinya terkoneksi. Tapi sepertinya ada masalah. Apakah ada penjelasan kenapa seperti ini, ya?

**Reply by kevbow** [reply]
Halo Pak Adrian,

Sepertinya pada bot_local.py Anda masih terdapat nilai yang kosong (pada template awal dituliskan dengan "___") yang Anda harus ubah terlebih dahulu. Mohon diperiksa kembali, selain itu kami belum dapat membantu lebih lanjut karena tidak dapat melakukan debug jika hanya menggunakan hasil foto. 
Anda dapat mengacu pada [halaman ini](https://github.com/teamalgoritma/community/discussions/29) tentang apa yang harus dilakukan ketika mengajukan pertanyaan agar lebih jelas. Tetapi karena informasi di panduan tersebut merupakan petunjuk bahasa R silakan Anda lampirkan kode yang kira-kira bermasalah agar kami dapat tinjau lebih lanjut. 
Sekian yang dapat kami sampaikan, terima kasih.

**Comment by adrianpratamasoe** [comment]
Template awalnya tidak ada ___ yg tidak terisi, sepertinya ada masalah ataukah ada sesuatu yg kurang pas
![image](https://user-images.githubusercontent.com/40779899/163664356-ed5c9014-4cad-4d0f-942b-19ec9eb87f47.png)
[](url
[bot_local.zip](https://github.com/teamalgoritma/community/files/8499394/bot_local.zip)
)
Saya kurang mengerti, kenapa ini terjadi

**Reply by kevbow** [reply]
Halo Pak Adrian,
Coba perhatikan kembali perintah pengerjaan temp.substitute(). Pastikan Anda menggunakan nama variable/parameter yang tepat di dalamnya disesuaikan dengan yang terdapat pada file welcome.txt. Sekian yang dapat kami sampaikan, terima kasih.

**Reply by adrianpratamasoe** [reply]
![image](https://user-images.githubusercontent.com/40779899/163737032-1b890a18-686e-4502-b8f1-7a2d24599190.png)
sepertinya masih error, kenapa ya?

**Reply by kevbow** [reply]
Dear Pak Adrian,

Jika Anda lihat pada welcome.txt terdapat kalimat berikut:
`Hello *${FULL_NAME}*, I can help you to provide a report on Facebook daily ads for different marketing campaigns.`

dimana ada sebuah variabel bernama FULL_NAME yang dimasukkan ke dalam kurung kurawal. Nama variabel inilah yang Anda masukkan ke dalam temp.substitute, sehingga kode yang seharusnya yaitu:
```python
welcome = temp.substitute(FULL_NAME = full_name)
```

Demikian dan terima kasih.

**Reply by adrianpratamasoe** [reply]
![image](https://user-images.githubusercontent.com/40779899/164617055-45359198-113f-47c4-b29d-a491697af63a.png)
Sepertinya masih error, walaupun sudah diperbaiki

**Comment by kevbow** [comment]
Halo Pak Adrian,

Sebelumnya apakah kode tersebut sudah aman ketika dijalankan di jupyter notebook?

**Reply by adrianpratamasoe** [reply]
Sudah mas, pas dijalankan di bot_local baru terjadi masalah
